File: Selenium/Activities/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions
with webdriver.Chrome() as driver:
    driver.get("https://v1.training-support.net/selenium/ajax")
    wait=WebDriverWait(driver,10)
    print("Home page title is :"+driver.title)
    driver.find_element(By.XPATH,"//button[text()='Change Content']").click()
    wait.until(expected_conditions.text_to_be_present_in_element((By.TAG_NAME,"h1"),"HELLO!"))
    helloText=driver.find_element(By.TAG_NAME,"h1").text
    print(helloText)
    ajaxText=driver.find_element(By.TAG_NAME,"h3")
    print(ajaxText.text)
    wait.until(expected_conditions.text_to_be_present_in_element((By.TAG_NAME,"h3"),"I'm late!"))
    print("after Ajax call loading the text is"+ajaxText.text)

File: Selenium/Activities/Activity9.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity9 {
    public static void main(String[] args) {
        WebDriver driver=new ChromeDriver();
        driver.get("https://v1.training-support.net/selenium/ajax");
        WebDriverWait wait=new WebDriverWait(driver, Duration.ofSeconds(10));
        System.out.println("Home page title is :"+driver.getTitle());
        driver.findElement(By.className("violet")).click();
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.tagName("h1")));
        // Find and print the new text
        String text = driver.findElement(By.tagName("h1")).getText();
        System.out.println(text);
        WebElement initialLateText=driver.findElement(By.tagName("h3"));
        System.out.println("Initially late text is :"+initialLateText.getText());

        //Wait for delayed text
        wait.until(ExpectedConditions.textToBePresentInElementLocated(By.tagName("h3"),"I'm late!"));
        WebElement ajaxLateText=driver.findElement(By.tagName("h3"));
        System.out.println("After Ajax is loaded the text is :"+ajaxLateText.getText());
        driver.close();
    }
    }

